lib/spack/spack/cmd/install.py

```diff
@@ -300,25 +310,25 @@ def install(parser, args, **kwargs):
     if args.no_checksum:
         spack.do_checksum = False        # TODO: remove this global.
 
+    only = set([x.strip() for x in args.only_str.split(',')])
```

Comment:
Minor thing: there's no need to turn the argument into a list. Invoking `__contains__` will succeed also with the plain string.